tonilopezmr/day6/MemoryBank.kt

```diff
@@ -0,0 +1,104 @@
+package day6
+
+import kategory.combine
+import org.junit.Assert.assertEquals
+import org.junit.Test
+
+typealias P = Pair<Int, Int>
+
+class MemoryBank {
+
+  private fun chooseBank(input: List<Int>): Pair<Int, Int> =
+      input.foldRightIndexed(P(0, 0)) { idx, bank, (index, blocks) ->
+        if (bank >= blocks) P(idx, bank) else P(index, blocks)
+      }
+
+  private fun balance(input: List<Int>): List<Int> {
+    var (index, bank) = chooseBank(input)
+    val distribution = if (bank < input.size) bank else input.size - 1
+    val blocksForEach = bank / distribution
+    val list = input.toMutableList()
+    val displacement = blocksForEach * distribution
+    list[index] = bank - displacement
+    index++
+    for (i in 1..distribution) {
+      index %= input.size
+      list[index] = list[index] + blocksForEach
+      index++
+    }
+
+    return list
+  }
```

Comment:
Simple enough, not accostumed to see that much mutability in Kotlin 😅

Reply:
xD that helps me understand the problem, you can see the "immutable" way also 👍